src/providers/keys/client.py

```diff
@@ -0,0 +1,51 @@
+from functools import lru_cache
+from time import sleep
+from typing import Optional
+
+from src.metrics.prometheus.basic import KEYS_API_REQUESTS_DURATION, KEYS_API_REQUESTS
+from src.providers.http_provider import HTTPProvider
+from src.providers.keys.typings import LidoKey, OperatorResponse
+from src.typings import BlockStamp
+from src.utils.dataclass import list_of_dataclasses
+
+
+class KeysOutdatedException(Exception):
+    pass
+
+
+class KeysAPIClient(HTTPProvider):
+    RETRY_COUNT = 10
+    REQUEST_TIMEOUT = 60
+    SLEEP_SECONDS = 10
+
+    PROMETHEUS_HISTOGRAM = KEYS_API_REQUESTS_DURATION
+    PROMETHEUS_COUNTER = KEYS_API_REQUESTS
+
+    ALL_KEYS = 'v1/keys'
+    ALL_OPERATORS = 'v1/operators'
+
+    def _get_with_blockstamp(self, url: str, blockstamp: BlockStamp, params: Optional[dict] = None) -> dict:
+        """
+        Returns response if blockstamp < blockNumber from response
+        """
+        for i in range(self.RETRY_COUNT):
+            data, meta = self._get(url, params)
+            if meta['meta']['elBlockSnapshot']['blockNumber'] >= blockstamp.block_number:
+                return data
```

Comment:
might be a good idea to check that total keys from Keys API >= total amount of keys that we fetches from staking router with `getAllNodeOperatorsReports` / `getAllNodeOperatorsDigests`